### DISCUSSION TITLE
Displayerweiterung um ST7567 Kontroller

### DISCUSSION THREAD

**Discussion by mat-sche** [tbnobody/OpenDTU#2296] [Q&A]
Hallo,

in der Vergangenheit hatte ich versucht in der Display_Graphic.h und angrenzenden  den Kontroller ST7567 mit einzubinden, was mir aber leider nicht gelang.
Ich nutze in einigen eigenen Projekten ebenfalls die LIB <U8g2lib.h> erfolgreich mit meinem Display und wollte mal fragen, ob jemand mir bei der Umsetzung/Erweiterung helfen könnte.

Grüße MAT

**Comment by ghost** [comment]
Hi,

ich habe auch schon mal ein zusätzliches Display, das 12864-59N / GM12864-59N, erfolgreich hinzu gefügt, seit einiger Zeit wird es offiziell unterstützt.
[Hier](https://github.com/tbnobody/OpenDTU/discussions/1641#discussion-6070050) ist meine Vorgehensweise beschrieben. Hat mangels Ahnung auch etwas gedauert, bis es bei mir geklappt hat. 
Die Zeilennummern könnten mittlerweile etwas anders sein.
Zu beachten wäre schlussendlich nur (wenn du ein neues Display hinzufügst); es wäre dann Typ = 6.

Wenn du zusätzlich eine evtl. vorhandene Hintergrundbeleuchtung mit anschließen und steuern willst, könnte ich auch dazu beitragen, ich müsste aber die Codeschnipsel erst heraussuchen; das könnte ein paar Tage dauern.
Alternativ kannst du auch in mein [Repository](https://github.com/schneeer/OpenDTU-Diagramm) schauen, da ist das schon alles drin. Ist aber nicht sonderlich gepflegt. Und, die aller letzten openDTU Neuerungen sind auch noch nicht eingeflossen.

@Entwickler: Einen Pin für die Hintergrundbeleuchtung hat das unterstützte Nokia Display auch.

**Reply by mat-sche** [reply]
Hi,

sorry für die späte Antwort.... Jetzt läuft ersteinmal das Vusions-Board mit openDTUon Battery. Ich werde es in 3 Wochen mal durchspielen und nochmals einpflegen. Mein LCD wird via SPI angesprochen, aktuell nutze ich bei meinen Cods in der Arduino-IDE  folgende Einstellungen:
U8G2_ST7565_JLX12864_F_4W_SW_SPI u8g2(U8G2_R2, /* clock=*/ 18, /* data=*/ 23, /* cs=*/ 5, /* dc=*/ 19, /* reset=*/ 14);

Die werde ich dann versuchen in der dtu mit einzubinden, vllt. auch als HW_SPI..
Vielen DAnk und bis später!

MAT
